src/rgw/rgw_asio_frontend.cc

```diff
@@ -175,6 +177,9 @@ void handle_connection(boost::asio::io_context& context,
       body.data = discard_buffer.data();
 
       http::async_read_some(stream, buffer, parser, yield[ec]);
+      if (ec == http::error::need_buffer) {
+        continue;
+      }
```

Comment:
was seeing `Broken pipe` failures from [test_encryption_sse_c_multipart_invalid_chunks_1](http://qa-proxy.ceph.com/teuthology/cbodley-2019-08-29_19:49:05-rgw-wip-cbodley-testing-distro-basic-smithi/4261818/teuthology.log) without this bit